http-client-javanet/src/main/java/io/micronaut/http/client/javanet/AbstractJavanetHttpClient.java

```diff
@@ -152,25 +150,28 @@ protected PasswordAuthentication getPasswordAuthentication() {
         return builder;
     }
 
-    private SSLContext configureSsl(ClientSslConfiguration clientSslConfiguration) {
-        try {
-            SSLContext sslContext = SSLContext.getInstance("TLS");
-            if (clientSslConfiguration.isInsecureTrustAllCertificates()) {
-                TrustManager[] trustAllCerts = new TrustManager[]{new TrustAllTrustManager()};
-                sslContext.init(null, trustAllCerts, null);
-            } else {
-                sslContext = SSLContext.getDefault();
-            }
-            return sslContext;
-        } catch (GeneralSecurityException e) {
-            throw new SslConfigurationException(e);
+    private void configureSsl(HttpClient.Builder builder, ClientSslConfiguration clientSslConfiguration) {
+        sslBuilder.build(clientSslConfiguration).ifPresent(builder::sslContext);
+        if (System.getProperty("jdk.internal.httpclient.disableHostnameVerification") != null && log.isWarnEnabled()) {
+            log.warn("The jdk.internal.httpclient.disableHostnameVerification system property is set. This is not recommended for production use.");
```

Comment:
also mention why it's not recommended

Reply:
https://github.com/micronaut-projects/micronaut-core/pull/8779/commits/48fd0a1491f3d4e09c61943e5a1be627369f0724